File: android/src/main/java/top/huic/tencent_im_plugin/message/ProfileSystemMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.TIMMessage;
import com.tencent.imsdk.TIMProfileSystemElem;

import top.huic.tencent_im_plugin.message.entity.ProfileSystemEntity;


/**
 * 用户资料变更系统通知 节点
 */
public class ProfileSystemMessageNode extends AbstractMessageNode<TIMProfileSystemElem, ProfileSystemEntity> {

    @Override
    protected TIMMessage getSendMessage(ProfileSystemEntity entity) {
        throw new RuntimeException("This node does not support sending");
    }

    @Override
    public String getNote(TIMProfileSystemElem elem) {
        return "[用户资料变更]";
    }

    @Override
    public ProfileSystemEntity analysis(TIMProfileSystemElem elem) {
        return new ProfileSystemEntity(elem);
    }

    @Override
    public Class<ProfileSystemEntity> getEntityClass() {
        return ProfileSystemEntity.class;
    }
}

File: ios/Classes/message/ProfileSystemMessageNode.swift
import ImSDK;

//
// Created by 蒋具宏 on 2020/7/11.
// 资料变更消息节点
class ProfileSystemMessageNode: AbstractMessageNode {
    override func getNote(elem: TIMElem) -> String {
        return "[用户资料变更]";
    }

    override func analysis(elem: TIMElem) -> AbstractMessageEntity {
        let profileSystemElem = elem as! TIMProfileSystemElem;
        return ProfileSystemMessageEntity(elem: profileSystemElem);
    }
}